File: 946-validate-stack-sequences/946-validate-stack-sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        j = 0
        st = []
        for i in pushed:
            st.append(i)
            while len(st) and st[-1] == popped[j]:
                st.pop()
                j+=1
        return j == len(popped)

File: 946-validate-stack-sequences/946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> st;
        int ele(0), sz(popped.size());
        for(int p : pushed)
        {
            st.push(p);
            while(st.size() && ele < sz && st.top() == popped[ele] )
            {
                st.pop();
                ele++;
            }
        }
        return ele == sz;
    }
};